x/rollapp/keeper/msg_server_update_state.go

```diff
@@ -34,17 +34,19 @@ func (k msgServer) UpdateState(goCtx context.Context, msg *types.MsgUpdateState)
 		return nil, errorsmod.Wrap(err, "before update state")
 	}
 
-	// verify the DRS version is not vulnerable
-	if k.IsDRSVersionVulnerable(ctx, msg.DrsVersion) {
-		// the rollapp is not marked as vulnerable yet, mark it now
-		err := k.MarkRollappAsVulnerable(ctx, msg.RollappId)
-		if err != nil {
-			return nil, fmt.Errorf("mark rollapp vulnerable: %w", err)
+	for _, bd := range msg.BDs.BD {
```

Comment:
the iteration here can be costly and potentially lead to dos as we currently don't limit the num blocks in a batch.
I think we can simply check the first and last block, cause he chances of DRS version changing in between is highly unlikely and even somehow it did, it's not that interesting to fork the rollap for.

Reply:
modified

Reply:
since the iteration does not imply any store access all bds are checked